### PR TITLE
[14.0][ADD] hr_attendance_sheet

### PR DESCRIPTION
**Pull request by Freni-OSI**

### CODE REVIEW

**Comment by Freni-OSI** [comment]
Needs Review. @patrickrwilson

**Review by patrickrwilson — APPROVED** [review]
Thanks Freni, LGTM

**Comment by patrickrwilson** [comment]
@osi-scampbell can you please review?

**Review by osi-scampbell — APPROVED** [review]

**Comment by Chanakya-OSI** [comment]
@Freni-OSI Can you Fix pre-commit and Travis

**Review by hkapatel-initos — CHANGES_REQUESTED** [review]
@Freni-OSI Can you please make pre-commit and Travis happy.

**Comment by hkapatel-initos** [comment]
@Freni-OSI Can you please make pre-commit and Travis happy.

**Comment by dsolanki-initos** [comment]
@Freni-OSI Can you make checks happy?

**Comment by dsolanki-initos** [comment]
@Freni-OSI Can you make checks happy?

**Comment by dsolanki-initos** [comment]
@Freni-OSI Can you make checks happy ASAP.

**Comment by dsolanki-initos** [comment]
@Freni-OSI could you please resolve the travis and pre-commit issue.

**Comment by max3903** [comment]
@Urvisha-OSI Please fix this PR

**Comment by Urvisha-OSI** [comment]
> @Urvisha-OSI Please fix this PR

@max3903 this PR pre-commit issue is fixed, there is codecov/patch is faild but this is because the is some lines ingnored by the test cases so there is not any issue you can merge this branch

**Review by max3903 — APPROVED** [review]

File: hr_attendance_sheet/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        "report/hr_attendance_sheet_report.xml",
+        "views/hr_attendance_sheet.xml",
+        "views/hr_attendance_view.xml",
+        "views/hr_department.xml",
+        "views/hr_employee.xml",
+        "views/res_config_settings_views.xml",
+        "views/res_company.xml",
+    ],
+    "development_status": "Beta",
+    "maintainer": "patrickrwilson",
```
Comment (max3903) [inline hr_attendance_sheet/__manifest__.py]:
```suggestion
    "maintainer": ["patrickrwilson", "max3903"],
```

**Comment by nimarosa** [comment]
Please rebase the PR and fix checks. Also maintainer key should be "maintainers". Probably this is what is causing checks to fail. 

Please rebase to the base branch and make sure checks passes so we can merge.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -46,7 +46,7 @@ repos:
           - --remove-duplicate-keys
           - --remove-unused-variables
   - repo: https://github.com/psf/black
-    rev: 20.8b1
+    rev: 22.3.0
```
Comment (nimarosa) [inline .pre-commit-config.yaml]: Why are you changing pre commit settings?

File: hr_attendance_sheet/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        "report/hr_attendance_sheet_report.xml",
+        "views/hr_attendance_sheet.xml",
+        "views/hr_attendance_view.xml",
+        "views/hr_department.xml",
+        "views/hr_employee.xml",
+        "views/res_config_settings_views.xml",
+        "views/res_company.xml",
+    ],
+    "development_status": "Beta",
+    "maintainer": ["patrickrwilson", "max3903"],
```
Comment (nimarosa) [inline hr_attendance_sheet/__manifest__.py]: The key should be "maintainers" not "maintainer"

**Review by nimarosa — CHANGES_REQUESTED** [review]

File: hr_attendance_sheet/__manifest__.py (hunk 3 above)
Comment (max3903) [inline hr_attendance_sheet/__manifest__.py]:
```suggestion
    "maintainers": ["patrickrwilson", "max3903"],
```

File: .pre-commit-config.yaml (hunk 2 above)
Comment (max3903) [inline .pre-commit-config.yaml]:
```suggestion
    rev: 20.8b1
```

**Comment by nimarosa** [comment]
@max3903 Please check tests failing

**Comment by max3903** [comment]
@Urvisha-OSI @Freni-OSI Can you fix this PR please?

**Comment by Freni-OSI** [comment]
To fix the error of pre-commit -ImportError: cannot import name '_unicodefun' from 'click', made changes in pre-commit config. please check the ref: https://github.com/psf/black/issues/2964
cc: @max3903 @nimarosa

**Comment by Urvisha-OSI** [comment]
> @Urvisha-OSI @Freni-OSI Can you fix this PR please?

@max3903 all checkes are passed you can check it

**Comment by max3903** [comment]
> @max3903 Please check tests failing

@nimarosa Done

**Pull request merged**